src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -127,31 +127,40 @@ protected void tick(Block b) {
                     if (!menu.fits(bucket.get(), getOutputSlots())) {
                         return;
                     }
-
+                    
+                    Block nextFluid = findNextFluid(fluid);
+                    if (nextFluid == null) return;
+              
                     removeCharge(b.getLocation(), ENERGY_CONSUMPTION);
                     menu.consumeItem(slot);
                     menu.pushItem(bucket.get().clone(), getOutputSlots());
-                    consumeFluid(fluid);
-
+                    nextFluid.setType(Material.AIR);
                     return;
                 }
             }
         }
     }
 
-    private void consumeFluid(Block fluid) {
-        if (fluid.isLiquid()) {
-            if (fluid.getType() == Material.WATER || fluid.getType() == Material.BUBBLE_COLUMN) {
-                // With water we can be sure to find an infinite source whenever we go
-                // further than 2 blocks, so we can just remove the water here and save
-                // outselves all of that computing...
-                fluid.setType(Material.AIR);
+    private Block findNextFluid(Block fluid) {
+        if (fluid.getType() == Material.WATER || fluid.getType() == Material.BUBBLE_COLUMN) {
+            // With water we can be sure to find an infinite source whenever we go
+            // further than 2 blocks, so we can just remove the water here and save
+            // ourselves all of that computing...
+            if (isSource(fluid)) {
+                return fluid;
             }
-            else {
-                List<Block> list = Vein.find(fluid, RANGE, block -> isLiquid(block) && block.getType() == fluid.getType());
-                list.get(list.size() - 1).setType(Material.AIR);
+            
+        } else if (fluid.getType() == Material.LAVA) {
```

Comment:
Please follow the existing formatting, the `else if` should go on the next line.

Reply:
Ugh none of the changes I contributed to the plugin had else's in next lines lol